scripts/zones/Ship_bound_for_Mhaura/Zone.lua

```diff
@@ -4,28 +4,95 @@
 --
 -----------------------------------
 local ID = require("scripts/zones/Ship_bound_for_Mhaura/IDs")
+require("scripts/globals/sea_creatures")
 -----------------------------------
 local zone_object = {}
 
 zone_object.onInitialize = function(zone)
 end
 
-zone_object.onZoneIn = function(player, prevZone)
+zone_object.onZoneIn = function(player, prevZone, zone)
     local cs = -1
-    if (player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0) then
+    local zoneID = 221
+
+    if
+        player:getXPos() == 0 and
+        player:getYPos() == 0 and
+        player:getZPos() == 0
+    then
         local position = math.random(-2, 2) + 0.150
         player:setPos(position, -2.100, 3.250, 64)
+        if
+            player:getGMLevel() == 0 and
+            GetZone(zoneID):getLocalVar('stateSet') == 0
+        then
+            GetZone(zoneID):setLocalVar('stateSet', 1)
+            GetZone(zoneID):setLocalVar('state', 2)
+            GetZone(zoneID):setLocalVar('transportTime', os.time())
+        end
     end
     return cs
 end
 
 zone_object.onTransportEvent = function(player, transport)
+    player:getZone():setLocalVar('stateSet', 0)
+    player:getZone():setLocalVar('state', 1)
     player:startEvent(512)
 end
 
 zone_object.onEventUpdate = function(player, csid, option)
 end
 
+zone_object.onGameHour = function()
+    local hour = VanadielHour()
+    if
+        hour >= 20 or
+        hour < 4
+    then
+        if
+            math.random() < 0.20 and
+            not GetMobByID(ID.mob.PHANTOM):isSpawned()
+        then
+            GetMobByID(ID.mob.PHANTOM):spawn()
+        end
+    elseif GetMobByID(ID.mob.PHANTOM):isSpawned() then
+        DespawnMob(ID.mob.PHANTOM)
+    end
+
+    local mob = GetMobByID(ID.mob.SEA_HORROR)
+    -- 3% chance per game hour (if not spawned, and min repop time)
+    if
+        math.random(0, 100) < 3 and
+        not mob:isSpawned() and
+        os.time() > mob:getLocalVar("respawnTime")
+    then
+        mob:spawn()
+    end
+end
+
+zone_object.onZoneTick = function(zone)
+    if zone:getLocalVar('state') == 1 then
```

Comment:
I mean could: 

0 is idle
1 is end
2 is start
3 is active

they don't match pirates.lua since the pirates enum only matter for 2/4 ship zones that use state. So I wanted to avoid using those perticular enums. I can make another enum for more descriptiveness in sea_creatures since its used in all four for the boat zone states.